Resources/doc/cookbook/recipe_hashid.rst

```diff
@@ -0,0 +1,50 @@
+Securing route id's with HashId
+===============================
+
+If you're working on an app where you need to hide the ID's from the routes then a good way to do ikt would be to use
+[HashId](http://hashids.org/). There is already a [Symfony Bundle](https://github.com/roukmoute/HashidsBundle) for this
+which you will need to install and configure.
+
+Overriding the output of all urls
+---------------------------------
+
+Encoding all the URL's that Sonata outputs is easy, see the example below
+
+```
+class MyAdmin extends AbstractAdmin
+{
+    ... other configurations
+    /**
+     * @param string $name
+     * @param mixed $object
+     * @param array $parameters
+     * @param bool|int $absolute
+     * @return string
+     */
+    public function generateObjectUrl($name, $object, array $parameters = [], $absolute = UrlGeneratorInterface::ABSOLUTE_PATH)
+    {
+        $parameters['id'] = $this->hashIDs->encodeHex($this->getUrlsafeIdentifier($object));
+
+        return $this->generateUrl($name, $parameters, $absolute);
+    }
+```
+
+Now your routes will change from /product/1/edit to /product/qf5/edit which mean that you need to change your
+controllers to decode these. There is a service that decodes the id from the request which is the `RouteIdHandler`,
+you would look to implement a version of the `RouteIdHandlerInterface` to decode the ID on the way in
+
+```
+class HashIdRouteIdHandler implements RouteIdHandlerInterface
+{
+    /**
+     * @param Request $request
+     * @param AdminInterface $admin
+     *
+     * @return int|string
+     */
+    public function getObjectId(Request $request, AdminInterface $admin)
+    {
+        return $hashId->decodeHex($request->get($admin->getIdParameter()));
+    }
+}
+```
```

Comment:
Ah, I've never written RST docs before, I will check the standard and work on the cookbook doc first so it will help understand the change that is needed.